makefiles/color.inc.mk

```diff
@@ -8,12 +8,8 @@ COLOR_RESET  :=
 COLOR_ECHO   := /bin/echo
 
 ifeq ($(CC_NOCOLOR),)
-  available_colors:=$(shell tput colors 2> /dev/null)
-  ifeq ($(available_colors),)
-    CC_NOCOLOR = 1
-  else ifeq ($(available_colors),-1)
-    CC_NOCOLOR = 1
-  else ifeq ($(available_colors),0)
+  IS_TERMINAL = $(if $(MAKE_TERMOUT),$(MAKE_TERMERR),)
```

Comment:
I can test this solves the issue you point out, but what happens if the terminal itlsed doesn't have multiple color support?

Reply:
It's up to the user if he is using a weird setup.

Reply:
In #8007 the code was introduced to check for those "weird setup". The code was working for that use case, the changes you are applying are not working for that use case, although they do fix your scenario.

If compiling with `TERM=at` in master you don't get color output which does happen with this PR. Could you modify your code so both checks are performed? i.e.:

```
ifeq ($(IS_TERMINAL),)
  CC_NOCOLOR = 1
else ifeq ($(available_colors),)
  CC_NOCOLOR = 1
else ifeq ($(available_colors),-1)
  CC_NOCOLOR = 1
else ifeq ($(available_colors),0)
  CC_NOCOLOR = 1
else
  CC_NOCOLOR = 0
endif
```

Reply:
what is `TERM=at`? what problem is this trying to solve? I believe that doing a fancy terminal type detection magic is out of scope for the RIOT build system.

Why should `TERM=at` disable color? I do `TERM=at ls --color=auto` and color still shows up.

This PR does something simple, than can be implemented without resorting to external commands: "disable colors by default if writing to a pipe". This is only for convenience, for reliability, setting CC_NOCOLOR is required.